src/test/java/org/onebusaway/alexa/AuthedSpeechletTest.java

```diff
@@ -278,6 +277,44 @@ public void help() throws SpeechletException {
     }
 
     @Test
+    public void repeat() throws SpeechletException, URISyntaxException, IOException {
+        IntentRequest repeatIntent = IntentRequest.builder()
+                .withRequestId("test-request-id")
+                .withIntent(
+                        Intent.builder()
+                                .withName("AMAZON.RepeatIntent")
+                                .withSlots(new HashMap<String, Slot>())
+                                .build()
+                )
+                .build();
+        // Try the repeat intent - this simulates after initial setup, when we don't yet have anything to repeat
+        SpeechletResponse sr = authedSpeechlet.onIntent(repeatIntent, session);
+        String spoken = ((PlainTextOutputSpeech)sr.getOutputSpeech()).getText();
+        assertThat(spoken, containsString("I'm sorry, I don't have anything to repeat.  You can ask me for arrival times for your stop."));
+
+        ObaArrivalInfo[] obaArrivalInfoArray = new ObaArrivalInfo[1];
+        obaArrivalInfoArray[0] = obaArrivalInfo;
+        new Expectations() {{
+            obaArrivalInfo.getShortName(); result = "8";
+            obaArrivalInfo.getHeadsign(); result = "Mlk Way Jr";
+            obaArrivalInfoResponse.getArrivalInfo(); result = obaArrivalInfoArray;
+            obaUserClient.getArrivalsAndDeparturesForStop(anyString, anyInt); result = obaArrivalInfoResponse;
+        }};
+        String response = "Route 8 Mlk Way Jr is now arriving based on the schedule -- ";
+
+        // Test initial request/response
+        sr = authedSpeechlet.onLaunch(launchRequest, session);
+        spoken = ((PlainTextOutputSpeech)sr.getOutputSpeech()).getText();
+        assertThat(spoken, equalTo(response));
+
+        // Now try repeat intent again - we should get the last response again
+        session.setAttribute(PREVIOUS_RESPONSE, response);  // FIXME - this should be set via the above call, but isn't?
```

Comment:
@philipmw Suggestions for how to fix this?  I expected that this session attribute should be set by making the above call to the `authedSpeechlet`, but it's not...